src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+import { get, uniq } from 'lodash-es';
+import { v4 as uuid } from 'uuid';
+
+import ConfigureDefaultField from './ConfigureDefaultField';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+import Alert from '../../../../components/Alert';
+import Type from './relationshipComponents/Type';
+
+function createCategory(relationships) {
+  const newRelationshipGuid = uuid();
+  return {
+    ...relationships,
+    [newRelationshipGuid]: { guid: newRelationshipGuid, label: '' },
+  };
+}
+
+function validateRelationships(relationships) {
+  let errors = [];
+  const types = Object.values(relationships);
+  const typeLabels = types.map(t => t?.label);
+  const emptyTypeLabels = typeLabels.filter(label => label === '');
+  if (emptyTypeLabels.length > 0)
+    errors.push('One or more relationship types are missing labels.');
+  const uniqueTypeLabels = uniq(typeLabels);
+  if (uniqueTypeLabels.length !== typeLabels.length)
+    errors.push(
+      'Two or more relationship types have the same label. Make sure each label is different.',
+    );
+  types.forEach(type => {
+    const typeLabel = type?.label;
+    const roleLabels = get(type, 'roles', []).map(r => r?.label);
+    const emptyRoleLabels = roleLabels.filter(label => label === '');
+    if (emptyRoleLabels.length > 0)
+      errors.push(
+        `One or more roles for the type ${typeLabel} are missing labels.`,
+      );
+    const uniqueRoleLabels = uniq(roleLabels);
+    if (uniqueRoleLabels.length !== roleLabels.length)
+      errors.push(
+        `Two or more roles for the type ${typeLabel} have the same label. Make sure each label is different.`,
+      );
+  });
+  return errors.length > 0 ? errors : null;
+}
+
+export default function RelationshipEditor({
+  onClose,
+  onSubmit,
+  formSettings,
+  setFormSettings,
+}) {
+  const [formErrors, setFormErrors] = useState(null);
+
+  function setRelationships(relationships) {
+    setFormSettings({ relationships });
```

Comment:
This should be `setFormSettings({ ...formSettings, relationships });`.

As @Atticus29 said earlier, it doesn't seem like this form state needs to be stored farther up the component tree than the `RelationshipEditor`. Maybe that is something to consider changing in the future.
>I'm guessing that this has something to do with all of them belonging to the same form schema. IMHO they don't need to be.

_Originally posted by @Atticus29 in https://github.com/WildMeOrg/codex-frontend/pull/282#discussion_r833771925_